worker/uniter/uniter.go

```diff
@@ -269,16 +269,21 @@ func (u *Uniter) loop(unitTag names.UnitTag) (err error) {
 	defer func() {
 		// If this is a CAAS unit, then dead errors are fairly normal ways to exit
 		// the uniter main loop, but the parent operator agent needs to keep running.
+		errorString := "<unknown>"
+		if err != nil {
+			errorString = err.Error()
+		}
 		if errors.Cause(err) == ErrCAASUnitDead {
 			err = nil
+			errorString = "caas unit dead"
 		}
 		if u.runListener != nil {
 			u.runListener.UnregisterRunner(unitTag.Id())
 		}
 		if u.localRunListener != nil {
 			u.localRunListener.UnregisterRunner(unitTag.Id())
 		}
-		u.logger.Infof("unit %q shutting down: %s", unitTag.Id(), err)
+		u.logger.Infof("unit %q shutting down: %s", unitTag.Id(), errorString)
```

Comment:
Should this be an Errorf if errorString is not empty?

Reply:
This will be triggered any time the loop exits, which might be just because the unit is being removed. So I don't think it is universally an error, which is why it is at INFO level.